docs/dev/adding_instruments.rst

```diff
@@ -327,3 +329,134 @@ The dictionary now maps the keys to specific values. The values and keys can be
     'Y'
 
 As you have seen, the :func:`Instrument.control <pymeasure.instruments.Instrument.control>` function can be significantly extended by using validators and maps.
+
+Processing of set values
+************************
+
+The :func:`Instrument.control <pymeasure.instruments.Instrument.control>`, and :func:`Instrument.setting <pymeasure.instruments.Instrument.setting>` allow a keyword argument `set_process` which must be a function that takes a value after validation and performs processing before value mapping. This function must return the processed value. This can be typically used for unit conversions as in the following example:
+
+
+.. testcode::
+
+    Extreme5000.current = Instrument.setting(
+        ":CURR %g",
+        """ A floating point property that takes the measurement current in A
+        """,
+        validator=strict_range,
+        values=[0, 10],
+        set_process=lambda v: 1e3*v  # convert current to mA
```

Comment:
```suggestion
        set_process=lambda v: 1e3*v,  # convert current to mA
```
Putting a trailing comma on the last kwarg like this keeps the diff smaller when modifying/adding kwargs, as this line does not need to be touched.